rfcs/0000-api-cr-crowding.md

```diff
@@ -0,0 +1,337 @@
+- If applicable, provide sample error messages, deprecation warnings, or migration guidance.
+- If applicable, describe the differences between teaching this to senior developers and to junior
+  developers.
+-->
+
+This introduces into the API the proposed new GTFS static resource called
+[occupancies.txt](https://github.com/google/transit/pull/240/files), which contains information
+about "usual or expected in-vehicle occupancy levels". The underlying data can apply to either a
+whole trip or individual stop times along the trip. A given occupancy specifies its
+`occupancy_status` which is an enum ranging from `0` for "Empty" to `6` for "Full". Similar to how
+we represent vehicles in the API, the API response will be the enum word rather than the number.
+Our feed will have an _extra_ field, currently not part of the spec, but which is being advocated
+for, `occupancy_percentage`, which is a value from `0.0` to `1.0`. Each occupancy has some
+validity constraints, namely the start and end dates and days of the week for which they're valid.
+A trip could therefore have several related occupancies.
+
+Our Keolis source data is at the trip level, and that is the only producer for which we will have
+static occupancy data in the near term. Initially, we will put this data directly in the API and
+not in our GTFS feed since the data updates as frequently as once/hour, and we don't want to
+update our GTFS static feed that often. However, we should ensure the API interface works _as if_
+it were coming from the GTFS feed, in case we do so in the future. While we're not given this
+data, we can arbitrarily say that the occupancies are valid for two weeks and apply to all days of
+the week. A weekend train trip generally doesn't run on weekdays, but if it were to (e.g. Labor
+Day running Sunday service), we'd want the predicted occupancy to be considered valid.
+
+Our API largely matches the GTFS feed and is simply a convenient way of accessing that data. While
+there are a few GTFS resources that we use in the API that are not exposed directly (e.g.
+transfers, calendar) and one major one that's named differently (schedule vs stoptime), the
+default expectation is that GTFS resources are available in the API.
+
+To that end, the API has a new `/occupancies` endpoint for an `Occupancy` resource. While the
+occupancy must have an ID – which we'll construct as `occupancy-[trip-id]` – there's no
+`/occupancies/[id]` endpoint, akin to `/predictions` or `/schedules`.
+
+For now, due to the experimental and malleable nature of the proposal, we want to start small and
+limit the coupling of occupancies with the rest of our code. So at a minimum we will provide just
+a single `/occupancies` endpoint that returns all current occupancies, with their trip
+relationship, like such:
+
+```
+GET /occupancies
+
+{
+  data: [
+    {
+      "attributes": {
+        "occupancy_status": MANY_SEATS_AVAILABLE,
+        "occupancy_percentage": 0.11,
```

Comment:
That's fair, but we use `occupancy_status` in other parts of the API / GTFS-RT.

Reply:
With the `occupancy_status` on `Vehicle` resources, I think the prefix makes more sense because the "occupancy" is not the resource the field is attached to (if it were just `vehicle.status` it would be ambiguous). AFAIK this is currently the only place occupancy data appears in the API.

Reply:
Ah, this is interesting. I initially just copied the name of the column from the GTFS proposal, and I do like that it's the same as on Vehicle.

However, this prompted me to look at how we do it elsewhere in the API, and it seems to be generally the case that the GTFS field repeats the resource name, but the API does not include it.

E.g:

## stops

**GTFS**
- stop_name
- stop_desc
- stop_lat
- stop_lon

**API**
- name
- description
- latitude
- longitude

## trips

**GTFS**
- trip_headsign
- trip_short_name

**API**
- headsign
- name

So I think in light of the precedent here, I'll omit the resource name from the field and go with `status` and `percentage`.